.github/workflows/autoformat.yml

```diff
@@ -0,0 +1,194 @@
+    if: ${{ needs.check-comment.outputs.matched == 'true' }}
+    outputs:
+      py_changed: ${{ steps.check-diff.outputs.py_changed }}
+      ui_changed: ${{ steps.check-diff.outputs.ui_changed }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+      - run: |
+          pip install requests
+      - name: Check diff
+        id: check-diff
+        run: |
+          repository="${{ needs.check-comment.outputs.repository }}"
+          pull_number="${{ needs.check-comment.outputs.pull_number }}"
+          changed_files="$(python dev/list_changed_files.py --repository $repository --pr-num $pull_number)"
+          py_changed=$([[ -z $(echo "$changed_files" | grep '\.py$') ]] && echo "false" || echo "true")
+          ui_changed=$([[ -z $(echo "$changed_files" | grep '^mlflow/server/js') ]] && echo "false" || echo "true")
+          echo "::set-output name=py_changed::$py_changed"
+          echo "::set-output name=ui_changed::$ui_changed"
+
+  python:
+    # Generate a patch to format python files.
+    runs-on: ubuntu-latest
+    needs: [check-comment, check-diff]
+    if: ${{ needs.check-diff.outputs.py_changed == 'true' }}
+    outputs:
+      has_diff: ${{ steps.check-diff.outputs.has_diff }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: ${{ needs.check-comment.outputs.repository }}
+          ref: ${{ needs.check-comment.outputs.ref }}
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.6'
+      - name: Install dependencies
+        run: |
+          pip install -r requirements/lint-requirements.txt
+      - name: Run black
+        run: |
+          black .
+      - name: Check diff
+        id: check-diff
+        run: |
+          git diff --output=python.diff
+          has_diff=$([[ -z "$(cat python.diff)" ]] && echo "false" || echo "true")
+          echo "::set-output name=has_diff::$has_diff"
+      - uses: actions/upload-artifact@v2
+        if: ${{ steps.check-diff.outputs.has_diff == 'true' }}
+        with:
+          name: python.${{ github.run_id }}.diff
+          path: python.diff
+          if-no-files-found: error
+          retention-days: 1
+
+  ui:
+    # Generate a patch to format files for MLflow UI.
+    runs-on: ubuntu-latest
+    needs: [check-comment, check-diff]
+    if: ${{ needs.check-diff.outputs.ui_changed == 'true' }}
+    outputs:
+      has_diff: ${{ steps.check-diff.outputs.has_diff }}
+    defaults:
+      run:
+        working-directory: mlflow/server/js
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: ${{ needs.check-comment.outputs.repository }}
+          ref: ${{ needs.check-comment.outputs.ref }}
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 10.x
+      - name: Install dependencies
+        run: |
+          npm i
+      - run: |
+          npm run lint:fix
```

Comment:
this is calling `.eslintrc.js` settings as defined in `package.json` operating solely on `mlflow.server.js.src`, right? So we're only looking at .js, .jsx, .ts, .tsx modules in src?
I'm likely completely wrong about this, but does jest's "collectCoverageFrom" apply to the `lint:fix` script execution to restrict auto-formatting lint corrections based on jest, or does it need to be explicitly defined in the lint:fix script config e.g., `"lint:fix": "eslint src --fix --ext .js,.jsx,.ts,.tsx"`?

Reply:
@BenWilson2 

> we're only looking at .js, .jsx, .ts, .tsx modules in src?

Yes.

> does jest's "collectCoverageFrom" apply to the lint:fix script execution

No, it doesn't.